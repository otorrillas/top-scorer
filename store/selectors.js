export const getGames = state => state.games;
