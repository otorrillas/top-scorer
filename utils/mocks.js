export const mockRuns = [
  {
    videos: { links: [{ uri: 'https://video-url' }] },
    players: [{ id: 'playerId' }],
    times: { primary_t: 450 }
  }
];

export const mockGames = {
  '1234': {
    abbreviation: 'test',
    assets: {
      'cover-medium': {
        uri: 'https://cover-medium-url'
      }
    },
    runs: mockRuns
  }
};

export const mockGamesResponse = [
  {
    id: '1234',
    ...mockGames['1234']
  },
  {
    id: '45678',
    ...mockGames['1234']
  }
];
