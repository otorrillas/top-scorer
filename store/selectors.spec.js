import { getGames } from './selectors';
import { mockGames } from 'Utils/mocks';

describe('Redux - selectors', () => {
  const state = {
    games: mockGames
  };

  describe('getGames', () => {
    test('returns mockGames', () => {
      expect(getGames(state)).toMatchSnapshot();
    });

    test('returns undefined on empty state', () => {
      expect(getGames({})).toMatchSnapshot();
    });
  });
});
