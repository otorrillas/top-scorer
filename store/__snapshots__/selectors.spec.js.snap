// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux - selectors getGames returns mockGames 1`] = `
Object {
  "1234": Object {
    "abbreviation": "test",
    "assets": Object {
      "cover-medium": Object {
        "uri": "https://cover-medium-url",
      },
    },
    "runs": Array [
      Object {
        "players": Array [
          Object {
            "id": "playerId",
          },
        ],
        "times": Object {
          "primary_t": 450,
        },
        "videos": Object {
          "links": Array [
            Object {
              "uri": "https://video-url",
            },
          ],
        },
      },
    ],
  },
}
`;

exports[`Redux - selectors getGames returns undefined on empty state 1`] = `undefined`;
